From ee42b38ef5698405aef1d4781a7852fabc5ae6d3 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Mon, 13 Jun 2016 14:08:46 +0800
Subject: [PATCH 1093/5242] MLK-12901-1 video: mipi_dsi_samsung: correct the
 hardware reset calling position.

commit  089d1700f4303480b8f9ba3e7e5b18f04641885f from
https://source.codeaurora.org/external/imx/linux-imx.git

The hardware reset should be done on LP-11 mode which
is the data/clk stop state.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/mipi_dsi_samsung.c |   23 ++++++++---------------
 1 file changed, 8 insertions(+), 15 deletions(-)

diff --git a/drivers/video/fbdev/mxc/mipi_dsi_samsung.c b/drivers/video/fbdev/mxc/mipi_dsi_samsung.c
index 2d52f84..38d1a44 100644
--- a/drivers/video/fbdev/mxc/mipi_dsi_samsung.c
+++ b/drivers/video/fbdev/mxc/mipi_dsi_samsung.c
@@ -548,14 +548,6 @@ static int mipi_dsi_enable(struct mxc_dispdrv_handle *disp,
 				return ret;
 			}
 		}
-
-		ret = device_reset(&mipi_dsi->pdev->dev);
-		if (ret) {
-			dev_err(&mipi_dsi->pdev->dev,
-				"failed to reset device: %d\n", ret);
-			return -EINVAL;
-		}
-		msleep(120);
 	}
 
 	if (!mipi_dsi->dsi_power_on)
@@ -578,6 +570,14 @@ static int mipi_dsi_enable(struct mxc_dispdrv_handle *disp,
 		 * in the dsi command mode.
 		 */
 		mipi_dsi_set_mode(mipi_dsi, DSI_COMMAND_MODE);
+		msleep(20);
+		ret = device_reset(&mipi_dsi->pdev->dev);
+		if (ret) {
+			dev_err(&mipi_dsi->pdev->dev, "failed to reset device: %d\n", ret);
+			return -EINVAL;
+		}
+		msleep(120);
+
 		ret = mipi_dsi->lcd_callback->mipi_lcd_setup(mipi_dsi);
 		if (ret < 0) {
 			dev_err(&mipi_dsi->pdev->dev,
@@ -781,12 +781,6 @@ static int mipi_dsi_probe(struct platform_device *pdev)
 		}
 	}
 
-	ret = device_reset(&pdev->dev);
-	if (ret) {
-		dev_err(&pdev->dev, "failed to reset device: %d\n", ret);
-		goto dev_reset_fail;
-	}
-
 	mipi_dsi->lcd_panel = kstrdup(lcd_panel, GFP_KERNEL);
 	if (!mipi_dsi->lcd_panel) {
 		dev_err(&pdev->dev, "failed to allocate lcd panel name\n");
@@ -816,7 +810,6 @@ static int mipi_dsi_probe(struct platform_device *pdev)
 dispdrv_reg_fail:
 	kfree(mipi_dsi->lcd_panel);
 kstrdup_fail:
-dev_reset_fail:
 	if (mipi_dsi->disp_power_on)
 		regulator_disable(mipi_dsi->disp_power_on);
 
-- 
1.7.9.5

